modules/openthread/Kconfig.features

```diff
@@ -130,6 +130,7 @@ config OPENTHREAD_DHCP6_SERVER
 
 config OPENTHREAD_DIAG
 	bool "Diagnostic functions support"
+	select NRF_802154_CARRIER_FUNCTIONS if IEEE802154_NRF5
```

Comment:
That's one way to achieve this, but wouldn't it actually make more sense to change the driver default, instead of making OT layer select certain feature of specific driver? Something like:
```
config NRF_802154_CARRIER_FUNCTIONS
	bool "nRF 802.15.4 carrier functions"
	default y if OPENTHREAD_DIAG
```